ichnaea/api/locate/tests/test_fallback.py

```diff
@@ -778,44 +826,35 @@ def test_cache_empty_result(self, geoip_db, http_session, session, source, stats
                 status_code=self.fallback_not_found_status,
             )
 
-            query = self.model_query(
-                geoip_db, http_session, session, stats, cells=[cell]
-            )
+            query = self.model_query(geoip_db, http_session, session, cells=[cell])
             results = source.search(query)
             self.check_model_results(results, None)
 
             assert mock_request.call_count == 1
-            stats.check(
-                counter=[
-                    ("locate.fallback.cache", [self.fallback_tag, "status:miss"]),
-                    (
-                        "locate.fallback.lookup",
-                        [
-                            self.fallback_tag,
-                            "status:%s" % self.fallback_not_found_status,
-                        ],
-                    ),
-                ]
+            assert metricsmock.has_record(
+                "incr",
+                "locate.fallback.cache",
+                value=1,
+                tags=[self.fallback_tag, "status:miss"],
             )
-
-            query = self.model_query(
-                geoip_db, http_session, session, stats, cells=[cell]
+            assert metricsmock.has_record(
+                "incr",
+                "locate.fallback.lookup",
+                value=1,
+                tags=[self.fallback_tag, "status:%s" % self.fallback_not_found_status],
             )
+            metricsmock.clear_records()
+
+            query = self.model_query(geoip_db, http_session, session, cells=[cell])
             results = source.search(query)
             self.check_model_results(results, None)
 
             assert mock_request.call_count == 1
-            stats.check(
-                counter=[
-                    ("locate.fallback.cache", [self.fallback_tag, "status:hit"]),
-                    (
-                        "locate.fallback.lookup",
-                        [
-                            self.fallback_tag,
-                            "status:%s" % self.fallback_not_found_status,
-                        ],
-                    ),
-                ]
+            assert metricsmock.has_record(
+                "incr",
+                "locate.fallback.cache",
+                value=1,
+                tags=[self.fallback_tag, "status:hit"],
```

Comment:
This is missing a check.

Reply:
My bad--this is a bug in the tests. The statsd state isn't getting cleared, so the second metric check is actually from the first set. When I add the `metricsmock.clear_records()`, then the second metric isn't there anymore, so we shouldn't be checking for it.